qiskit/quantum_info/synthesis/qsd.py

```diff
@@ -71,7 +82,18 @@ def qs_decomposition(mat, opt_a1=True, decomposer_1q=None, decomposer_2q=None):
         circ = decomposer_1q(mat)
     elif dim == 4:
         if decomposer_2q is None:
-            decomposer_2q = two_qubit_decompose.TwoQubitBasisDecomposer(CXGate())
+            if opt_a2:
+                from qiskit.extensions.unitary import UnitaryGate  # pylint: disable=cyclic-import
+
+                def decomp_2q(mat):
+                    ugate = UnitaryGate(mat)
+                    qc = QuantumCircuit(2, name="qsd2q")
+                    qc.append(ugate, [0, 1])
+                    return qc
+
+                decomposer_2q = decomp_2q
+            else:
+                decomposer_2q = two_qubit_decompose.TwoQubitBasisDecomposer(CXGate())
```

Comment:
Is there a reason to not just use `two_qubit_decompose.two_qubit_decompose` here? It's already initialized as `TwoQubitBasisDecomposer(CXGate())` (although wrapped in a lazy loader).

Reply:
maybe? mentioned above

Reply:
I think at one time I had an option to use a different two qubit gate but that seems to be gone. This looks like a fine use of `two_qubit_decompose`. Added in 8ee787.